controllers/cinder_controller.go

```diff
@@ -845,6 +845,24 @@ func (r *CinderReconciler) transportURLCreateOrUpdate(
 }
 
 func (r *CinderReconciler) apiDeploymentCreateOrUpdate(ctx context.Context, instance *cinderv1beta1.Cinder) (*cinderv1beta1.CinderAPI, controllerutil.OperationResult, error) {
+
+	apiTemplate := instance.Spec.CinderAPI
+	cinderAPISpec := cinderv1beta1.CinderAPISpec{
+		ContainerImage:         apiTemplate.ContainerImage,
+		Replicas:               apiTemplate.Replicas,
+		Debug:                  apiTemplate.Debug,
+		CustomServiceConfig:    apiTemplate.CustomServiceConfig,
+		DefaultConfigOverwrite: apiTemplate.DefaultConfigOverwrite,
+		NetworkAttachments:     apiTemplate.NetworkAttachments,
+		ServiceUser:            instance.Spec.ServiceUser,
+		DatabaseHostname:       instance.Status.DatabaseHostname,
+		DatabaseUser:           instance.Spec.DatabaseUser,
+		Secret:                 instance.Spec.Secret,
+		PasswordSelectors:      instance.Spec.PasswordSelectors,
+		ExtraMounts:            instance.Spec.ExtraMounts,
```

Comment:
This confuses me. Won't you end up overwriting the service's own list of ExtraMounts with the top level CR's ExtraMounts?

Reply:
Thanks Alan for this comment: let me try to clarify this part (we can follow up offline in case we want to dig more).
For the single service (e.g. `Cinder{Api, Sched, Bak, Vol}`) we don't have to specify/override `ExtraVolumes`: we
 can always define them in the (`Cinder`) top-level `CR` and _"Propagate"_ them to the single service (it's what
 we called **"internal"** propagation). 
 Here's an example [1], where we can define two `extraVol`, one is destined to `Cinder`, which means it will be
 propagated to all the `Cinder` related pods/services, the other one is only destined to `volume1` instance and all
 the `CinderBackup` related pods.
 
 
**tl;dr** given the described approach, we only define `extraMounts` in the top-level `Cinder` `CR`:  each service
 inherits the struct from the "upper  layer" and filter the volumes according to the `Propagation` policy. The code
 change here doesn't change at all the previous behavior (look at [2], which is the same for API/Sched/Back/Vol).

[1] https://paste.opendev.org/show/bLQMPvZL3IfSzYFFN2Hh/
[2] https://github.com/fmount/cinder-operator/blob/master/controllers/cinder_controller.go#L864

Reply:
Thanks, this explanation really helped. I had forgotten the whole ExtraVolumes concept includes its own mechanism for propagating volumes to specific CRs. This is why you define your ExtraVolumes at the top, and copy them into the lower layers.